File: Samples/ZeroDarkTodo/ZeroDarkTodo/UI tools/IconTitleButton.m
//
//  IconTitleButton.m
//  storm4_iOS
//
//  Created by Robbie Hanson on 11/8/17.
//  Copyright © 2017 4th-A Technologies, LLC. All rights reserved.
//

#import "IconTitleButton.h"

@implementation IconTitleButton

+ (instancetype)buttonWithType:(UIButtonType)buttonType
{
	IconTitleButton *button = [super buttonWithType:buttonType];
	[button postButtonWithTypeInit];
	return button;
}

/// Because we can't override init on a uibutton, do init steps here.
- (void)postButtonWithTypeInit
{
	self.frame = CGRectMake(0, 0, 200, 44);
	self.titleLabel.numberOfLines = 1;
	self.titleLabel.adjustsFontSizeToFitWidth = YES;
	self.titleLabel.lineBreakMode = NSLineBreakByClipping; //  MAGIC LINE
}

- (void)layoutSubviews
{
	[super layoutSubviews];
	self.titleEdgeInsets = UIEdgeInsetsMake(0.0, 8, 0, 0); // top, left, bottom, right
}

- (void)setImage:(UIImage *)image forState:(UIControlState)state
{
	[super setImage:image forState:state];
	self.imageView.layer.cornerRadius = image.size.width /2;
	self.imageView.layer.masksToBounds = YES;
}

@end

File: Samples/ZeroDarkTodo/ZeroDarkTodo/UI tools/IconTitleButton.swift
/// ZeroDark.cloud
///
/// Homepage      : https://www.zerodark.cloud
/// GitHub        : https://github.com/4th-ATechnologies/ZeroDark.cloud
/// Documentation : https://zerodarkcloud.readthedocs.io/en/latest
/// API Reference : https://apis.zerodark.cloud
///
/// Sample App: WhatsZapp

import UIKit

class IconTitleButton: UIButton {
	
	class func create() -> IconTitleButton {
		
		let button = IconTitleButton.init(type: .custom)
		
		button.heightAnchor.constraint(equalToConstant: 44).isActive = true
		
		button.titleLabel?.numberOfLines = 1
		button.titleLabel?.adjustsFontSizeToFitWidth = true
		button.titleLabel?.lineBreakMode = .byClipping //  MAGIC LINE
		
		return button
	}
	
	override func layoutSubviews() {
		super.layoutSubviews()
		self.titleEdgeInsets = UIEdgeInsets(top: 0, left: 8, bottom: 0, right: 0)
	}
	
	override func setImage(_ image: UIImage?, for state: UIControl.State) {
		super.setImage(image, for: state)
		if let image = image {
			self.imageView?.layer.cornerRadius = image.size.width / 2.0
			self.imageView?.layer.masksToBounds = true
		}
	}

}
